pyteal/ast/gitxn.py

```diff
@@ -57,18 +58,23 @@ def __str__(self):
 
     def __teal__(self, options: "CompileOptions"):
         verifyFieldVersion(self.field.arg_name, self.field.min_version, options.version)
-        if type(self.txnIndex) is not int or type(self.index) is not int:
```

Comment:
This should still raise an error if `self.txnIndex` is an int, since we don't have a version of `gitxn` that takes transaction indexes from the stack yet

Reply:
@jasonpaulos Thanks for catching the runtime type check mistake and the unnecessary cast.  
* I attempted to fix both issues via https://github.com/algorand/pyteal/pull/193/commits/57725e0812577bbbfb4720c3271617f3ea6700dc.
* Pointers appreciated if I'm doing something in a _non-Pythonic_ way.  I'm trying to balance researching the language while getting something to work.

Aside:   What conventions (if any) exist for unit tests confirming an exception does _not_ happen?
* I ran out of time to look closely today.  A quick google suggests writing a test _without_ assertions like https://stackoverflow.com/a/20275035.
* `gitxn_test.py` seems to apply the _test without assertion_ pattern, so I extended it.  Though it's unclear to me if it's preferable.

Reply:
57725e0812577bbbfb4720c3271617f3ea6700dc looks good, thanks for addressing my comments. And yes, we tend to use the test with assertions approach if the only goal is to make sure something does not raise an error.

Your unit test made me realized there's a hole in our type checking, which I elaborated more on in another comment.